pipe_segment/pipeline.py

```diff
@@ -60,13 +64,27 @@ def time_bin_key(x, time_bins):
 
 
 class SegmentPipeline:
-    def __init__(self, options):
-        self.cloud_options = options.view_as(GoogleCloudOptions)
-        self.options = options.view_as(SegmentOptions)
+    def __init__(self, options, beam_options):
+        self.options = options
+        self.beam_options = beam_options
+        self.cloud_options = beam_options.view_as(GoogleCloudOptions)
         self.date_range = parse_date_range(self.options.date_range)
         self.satellite_offsets_writer = None
         self.bqtools = BigQueryTools(project=self.cloud_options.project)
 
+        if self.options.sat_offset_dest:
+            remove_satellite_offsets_content(
+                options.sat_offset_dest,
+                options.date_range,
+                self.cloud_options.labels,
+                self.cloud_options.project)
```

Comment:
This wasn't being called?

Reply:
This `remove_satellite_offsets_content` is one of the things that are interacting with the database **before** the pipeline starts running.